src/bin/main.rs

```diff
@@ -81,14 +81,14 @@ fn run (args: Args) -> Result<(), Error> {
 
     // first we declare our symbolic references;
     // it is a runtime error to define a symbol _without_ declaring it first
-    let declarations = vec![
-        ("deadbeef",   Decl::Function { global: false }),
-        ("main",       Decl::Function { global: true }),
-        ("str.1",      Decl::CString { global: false }),
-        ("DEADBEEF",   Decl::DataImport),
-        ("STATIC",     Decl::Data { global: true, writable: true }),
-        ("STATIC_REF", Decl::Data { global: true, writable: true }),
-        ("printf",     Decl::FunctionImport),
+    let declarations: Vec<(&'static str, Decl)> = vec![
+        ("deadbeef", Decl::function().into()),
+        ("main", Decl::function().global().into()),
+        ("str.1", Decl::cstring().into()),
+        ("DEADBEEF", Decl::data_import().into()),
+        ("STATIC", Decl::data().global().writable().into()),
+        ("STATIC_REF", Decl::data().global().writable().into()),
+        ("printf", Decl::function_import().into()),
```

Comment:
function().import() seems more elegant; ditto for data_import, but maybe that’s impractical ?

Reply:
There's no reason we couldn't do that, except that I believe imports and declarations should be two different things in the API.